pkg/generator.go

```diff
@@ -699,6 +703,32 @@ func (_c *{{.CallStruct}}) Return({{range .Returns.Params}}{{.}},{{end}}) *{{.Ca
 `)
 }
 
+func (g *Generator) generateConstructor() {
+	const constructor = `
+// %[1]s creates a new instance of %[2]s. It also registers a cleanup function to assert the mocks expectations.
+func %[1]s(t testing.TB) *%[2]s {
+	mock := &%[2]s{}
+
+	t.Cleanup(func() { mock.AssertExpectations(t) })
+
+	return mock
+}
+`
+
+	funcFirstLetter := "N"
+	mockName := g.mockName()
+
+	for _, firstLetter := range mockName {
+		if unicode.IsLower(firstLetter) {
+			funcFirstLetter = "n"
```

Comment:
I've seen a repeated pattern of where we manually modify the first letter depending on whether it's exported or not. Might make sense have a helper function that will make a string exported/non-exported depending on the state of some other variable/function name.

Reply:
I didn't but yeah if you know about such places it would make sense to make a helper function :) I know about the `upperFirstOnly`, but it always changes the string (doesn't check another).

Reply:
If you could create a helper function in a new pkg/strings.go or whatever name makes sense, that would be great. I'm sure we'll use it again. Thanks!

Reply:
See this: https://github.com/vektra/mockery/pull/452